nilearn/maskers/surface_maps_masker.py

```diff
@@ -259,45 +259,27 @@ def transform(self, img, confounds=None, sample_mask=None):
                   default=None
             sample_mask to pass to :func:`nilearn.signal.clean`.
 
-
         Returns
         -------
         region_signals: :obj:`numpy.ndarray`
             Signal for each region as provided in the maps (via `maps_img`).
             shape: (n_timepoints, n_regions)
         """
-        check_is_fitted(self)
+        check_compatibility_mask_and_images(self.maps_img, imgs)
+        check_same_n_vertices(self.maps_img.mesh, imgs.mesh)
 
-        check_compatibility_mask_and_images(self.maps_img, img)
-        # if img is a single image, convert it to a list
-        # to be able to concatenate it
-        if not isinstance(img, list):
-            img = [img]
-        img = concat_imgs(img)
         # check img data is 2D
-        img.data._check_ndims(2, "img")
-        check_same_n_vertices(self.maps_img.mesh, img.mesh)
+        imgs.data._check_ndims(2, "img")
```

Comment:
could `at_least_2d` be used here?

Reply:
There are still common parts of `transform_single_imgs` for both estimators. I do not know if it would the case of other estimator implementations, but it might worth considering common parts to extract in another method not to repeat in each inherited class.